code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm

```diff
@@ -71,13 +71,8 @@
 
 /obj/machinery/atmospherics/components/binary/crystallizer/update_overlays()
 	. = ..()
-	cut_overlays()
-	var/mutable_appearance/pipe_appearance1 = mutable_appearance('icons/obj/pipes_n_cables/pipe_underlays.dmi', "intact_[dir]_[piping_layer]", layer = GAS_SCRUBBER_LAYER)
-	pipe_appearance1.color = COLOR_LIME
-	var/mutable_appearance/pipe_appearance2 = mutable_appearance('icons/obj/pipes_n_cables/pipe_underlays.dmi', "intact_[REVERSE_DIR(dir)]_[piping_layer]", layer = GAS_SCRUBBER_LAYER)
-	pipe_appearance2.color = COLOR_MOSTLY_PURE_RED
-	. += pipe_appearance1
-	. += pipe_appearance2
+	. += get_pipe_image(icon, "pipe", dir, COLOR_LIME, piping_layer)
+	. += get_pipe_image(icon, "pipe", REVERSE_DIR(dir), COLOR_MOSTLY_PURE_RED, piping_layer)
```

Comment:
do we just not need these sprites? why were we using them before if we don't care about them now?

Reply:
those sprites are still used in drawing the underlays when the machine is connected to something, which is all handled in `/obj/machinery/atmospherics/components/update_icon`, I'm not sure why they were used here to do the unconnected underlays because they're not the right size for the sprite (machines.dmi has a special 32x64 pipe sprite that aligns properly)

Reply:
actually if the connected underlays work then those sprites should be working, I think it might be the cut overlays proc that's breaking it

Reply:
yeah it was